client/client.go

```diff
@@ -130,50 +115,53 @@ func makeClient(cfg clientConfig) (Client, error) {
 }
 
 type clientConfig struct {
-	// URLs when specified will create an HTTP client.
-	urls []string
-	// Insecure will allow creating the HTTP client without a bound group.
-	insecure bool
+	// clients is the set of options for fetching randomness
+	clients []Client
+	// watcher is a constructor function for generating a new partial client of randomness
+	watcher WatcherCtor
 	// from `chainInfo.Hash()` - serves as a root of trust for a given
 	// randomness chain.
 	chainHash []byte
 	// Full chain information - serves as a root of trust.
 	chainInfo *chain.Info
-	// transport configures the http parameters used when fetching randomness.
-	transport http.RoundTripper
+	// insecure indicates the root of trust does not need to be present.
+	insecure bool
 	// cache size - how large of a cache to keep locally.
 	cacheSize int
 	// customized client log.
 	log log.Logger
 	// time after which a watcher will failover to using client.Get to get the latest randomness.
 	failoverGracePeriod time.Duration
-	// watcher is a constructor function that creates a new Watcher
-	watcher WatcherCtor
 	// autoWatch causes the client to start watching immediately in the background so that new randomness is proactively fetched and added to the cache.
 	autoWatch bool
 	// prometheus is an interface to a Prometheus system
 	prometheus prometheus.Registerer
 }
 
+func (c *clientConfig) tryPopulateInfo(cli Client) {
+	if c.chainInfo == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+		defer cancel()
+		c.chainInfo, _ = cli.Info(ctx)
+	}
+}
+
 // Option is an option configuring a client.
 type Option func(cfg *clientConfig) error
 
-// WithHTTPEndpoints configures the client to use the provided URLs.
-func WithHTTPEndpoints(urls []string) Option {
+// From constructs the client from a set of clients providing randomness
+func From(c ...Client) Option {
```

Comment:
Any particular reason not to make this method a constructor already ? That avoids one more call to `client.New` at the "cost" of having a slice of clients instead of a varargs.

Reply:
At the same time, it's also elegant to just chain option without resorting to `[]client.Client{...}` -

Reply:
can you do `client.From(clients ...Client, opt ...Option)`? I don't think you can have multiple variadic parameters?

Reply:
No you can't that's why I said the cost is to use a slice instead of varargs for the clients argument.

Reply:
TBH im not sure I have a preference for either :D

Reply:
added a `Wrap` convenience method. Good enough?